### PR TITLE
feat: add new Markdown scalar type

### PR DESCRIPTION
**Pull request by RutZap**
Introduce a new built-in field type: `Markdown`. Its behaviour is consistent with `Text` but by differentiating it from `Text` allows api consumers to make use of rich text editors.

### CODE REVIEW

File: schema/validation/rules/expression/expression.go [hunk 1]
```diff
@@ -286,6 +286,7 @@ func OperandTypesMatchRule(asts []*parser.AST, condition *parser.Condition, cont
 	// Possibly this only applies to Date and Timestamp
 	comparable := [][]string{
 		{parser.FieldTypeDate, parser.FieldTypeDatetime},
+		{parser.FieldTypeMarkdown, parser.FieldTypeText},
```
Comment (davenewza) [inline schema/validation/rules/expression/expression.go:289]: Nice

**Review by davenewza — APPROVED** [review]
Looks great.  That's a lot of tests!

File: integration/testdata/built_in_actions/schema.keel [hunk 2]
```diff
@@ -4,12 +4,13 @@ model Post {
...
 
     actions {
-        create createPost() with (title, subTitle)
+        create createPost() with (title, subTitle, content?)
         get getPost(id)
-        update updatePost(id) with (title)
+        update updatePost(id) with (title, content?)
         delete deletePost(id)
         delete deletePostBySubTitle(subTitle)
         list listPosts(id?, title?, category?, rating?)
```
Comment (davenewza) [inline integration/testdata/built_in_actions/schema.keel:16]: I would have probably added one implicit filtering test and one expression filtering test with `list`

File: runtime/actions/query_test.go [hunk 3]
```diff
@@ -510,12 +512,14 @@ var testCases = []testCase{
...
 					age Number
 					isActive Boolean
+					bio Markdown
 				}
 				actions {
 					update updatePerson(id) {
 						@set(person.name = "Bob")
 						@set(person.age = 100)
 						@set(person.isActive = true)
+						@set(person.bio = "## Biography")
```
Comment (adaam2) [inline runtime/actions/query_test.go:522]: What about a test for markdown contain quotes here?

File: runtime/jsonschema/jsonschema.go [hunk 4]
```diff
@@ -395,7 +395,7 @@ func jsonSchemaForField(ctx context.Context, schema *proto.Schema, action *proto
 			prop = JSONSchema{OneOf: oneOf}
 		}
 
-	case proto.Type_TYPE_ID, proto.Type_TYPE_STRING:
+	case proto.Type_TYPE_ID, proto.Type_TYPE_STRING, proto.Type_TYPE_MARKDOWN:
```
Comment (louisaspicer) [inline runtime/jsonschema/jsonschema.go:398]: Btw this is where we should change it to another case and have 
```
    prop.Type = "string"
    prop.Format = "markdown"
```

**Pull request merged**